coremods/src/main/java/net/neoforged/neoforge/coremods/MethodRedirector.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.coremods;
+
+import cpw.mods.modlauncher.api.ITransformer;
+import cpw.mods.modlauncher.api.ITransformerVotingContext;
+import cpw.mods.modlauncher.api.TargetType;
+import cpw.mods.modlauncher.api.TransformerVoteResult;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.tree.ClassNode;
+import org.objectweb.asm.tree.MethodInsnNode;
+
+/**
+ * Redirect calls to one method to another.
+ */
+public class MethodRedirector implements ITransformer<ClassNode> {
+    private final Map<String, List<MethodRedirection>> redirectionsByClass = new HashMap<>();
+    private final Set<Target<ClassNode>> targets = new HashSet<>();
+
+    private static final List<MethodRedirection> REDIRECTIONS = List.of(
```

Comment:
This is a weird way to do it that's inconsistent with `ReplaceFieldWithGetterAccess`. I would expand the method redirector ctor with `MethodRedirection` components and make the transformer per-redirection instead of global.

Reply:
I'm not so sure it's necessarily needed, considering there's only one anyway.